openmc/tallies.py

```diff
@@ -0,0 +1,3204 @@
+                    break
+
+        if nuclide_index == -1:
+            msg = 'Unable to get the nuclide index for Tally since "{0}" ' \
+                  'is not one of the nuclides'.format(nuclide)
+            raise KeyError(msg)
+        else:
+            return nuclide_index
+
+    def get_score_index(self, score):
+        """Returns the index in the Tally's results array for a score bin
+
+        Parameters
+        ----------
+        score : str
+            The score string (e.g., 'absorption', 'nu-fission')
+
+        Returns
+        -------
+        score_index : int
+            The index in the Tally data array for this score.
+
+        Raises
+        ------
+        ValueError
+            When the argument passed to the 'score' parameter cannot be found in
+            the Tally.
+
+        """
+
+        try:
+            score_index = self.scores.index(score)
+
+        except ValueError:
+            msg = 'Unable to get the score index for Tally since "{0}" ' \
+                  'is not one of the scores'.format(score)
+            raise ValueError(msg)
+
+        return score_index
+
+    def get_filter_indices(self, filters=[], filter_bins=[]):
+        """Get indices into the filter axis of this tally's data arrays.
+
+        This is a helper method for the Tally.get_values(...) method to
+        extract tally data. This method returns the indices into the filter
+        axis of the tally's data array (axis=0) for particular combinations
+        of filters and their corresponding bins.
+
+        Parameters
+        ----------
+        filters : Iterable of openmc.FilterMeta
+            An iterable of filter types
+            (e.g., [MeshFilter, EnergyFilter]; default is [])
+        filter_bins : Iterable of tuple
+            A list of tuples of filter bins corresponding to the filter_types
+            parameter (e.g., [(1,), ((0., 0.625e-6),)]; default is []). Each
+            tuple contains bins for the corresponding filter type in the filters
+            parameter. Each bin is an integer ID for Material-, Surface-,
+            Cell-, Cellborn-, and Universe- Filters. Each bin is an integer
+            for the cell instance ID for DistribcellFilters. Each bin is a
+            2-tuple of floats for Energy- and Energyout- Filters corresponding
+            to the energy boundaries of the bin of interest. The bin is an
+            (x,y,z) 3-tuple for MeshFilters corresponding to the mesh cell
+            of interest. The order of the bins in the list must correspond to
+            the filter_types parameter.
+
+        Returns
+        -------
+        numpy.ndarray
+            A NumPy array of the filter indices
+
+        """
+
+        cv.check_type('filters', filters, Iterable, openmc.FilterMeta)
+        cv.check_type('filter_bins', filter_bins, Iterable, tuple)
+
+        # Determine the score indices from any of the requested scores
+        if filters:
```

Comment:
we should have the main logic outside of all code blocks. This is because if blocks should be exceptions to the normal flow of the entire function.
Let's switch the if statement around to support this:
```
if not filters:
  filter_indices = np.arange(self.num_filter_bins)
  return filter_indices

*** rest of if block here ***
```